aten/src/ATen/core/BackendSelectFallbackKernel.cpp

```diff
@@ -6,7 +6,7 @@ using c10::KernelFunction;
 
 namespace {
 
-static auto registry = Dispatcher::singleton().registerBackendFallbackKernel(
+static auto registry = Dispatcher::singleton().registerFallback(
```

Comment:
Why remove the "backend" from the name? I find it clearer to specify it, especially since we also have fallthrough and we had (or actually still have) operator level fallbacks (i.e. compound ops) too.

Reply:
Two reasons:
1. Backend is not a concept that we've actually deified as a first class concept. We have plenty of "middle-end" dispatch keys which aren't actually the terminus of the chain. IIRC, "at::Backend" is still something we want to get rid of. So we shouldn't put "Backend" in the name here; at the very least, you'd have to come up with something else
2. Re confusion with other similar sounding concepts, I think the risk of confusion with operator level fallback is low, as the other two concepts integrate with the API in orthogonal ways (fallthrough is a way to implement a kernel, whereas fallback is a way you register a kernel; compound ops are the absence of a dispatch key).
3. I think Backend looks ugly and I'd prefer if all the method names are single word

Reply:
FWIW agree with Ed on (1) and (3), and would put some real weight behind the second part of (3) - especially in an API that relies (and might continue to) on chaining.

BUT to belabor (2), it's definitely super key for us to assign distinct, canonical names to these various concepts. Even just reading this thread I got confused again. 

Here are some terms we're using:
1. fallback
2. fallthrough
3. compound
4. catchAll

Personally I'd rather see us find more evocative short words and avoid e.g. prepending "backend" to (1). 

Just to recap what I think 1-4 mean, but please check:

1. a dispatch key-specific, but not op-specific, kernel the dispatcher invokes *in preference to* any op-specific kernels, when the dispatch key is matched. (A "column fill" in the dispatch table.)

2. a no-op kernel, used in conjunction with (1) and the reentrancy of the dispatcher to implement (let's call them here) hooks using dispatch keys, like BackendSelect.

3. A "row fill" in the dispatch table - per Ed above, a kernel registered for an op, but without a dispatch key. 

4. Is this the same as compound??

Reply:
Yes this is correct.

Reply:
My vote would be to go with `fallback` here and commit to having a canonical glossary as part of the finished new API

Reply:
I read too quickly.

> a dispatch key-specific, but not op-specific, kernel the dispatcher invokes in preference to

No, it's the opposite way. Op kernels always are taken first, only if there is no op kernel, do we go to the fallback.